MSAL/src/native_auth/public/state_machine/state/SignInAfterResetPasswordState.swift

```diff
@@ -0,0 +1,49 @@
+//
+// Copyright (c) Microsoft Corporation.
+// All rights reserved.
+//
+// This code is licensed under the MIT License.
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files(the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and / or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions :
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+
+import Foundation
+
+/// An object of this type is created when a user has reset their password successfully.
+@objcMembers public class SignInAfterResetPasswordState: SignInAfterPreviousFlowBaseState {
+    /// Sign in the user that signed up.
+    /// - Parameters:
+    ///   - scopes: Optional. Permissions you want included in the access token received after sign in flow has completed.
+    ///   - delegate: Delegate that receives callbacks for the Sign In flow.
+    public func signIn(scopes: [String]? = nil, delegate: SignInAfterResetPasswordDelegate) {
```

Comment:
Hey Silviu, maybe I didn't fully get your question. Can you please clarify?
Scopes are accepted only during signIn (when we're requesting an access token), and not during signUp or reset password.

Reply:
Indeed, I was under the impression that on SignUp we also specify scopes and it carries all the way to sign in via the continuation token. But now I realised they we don't have that capacity. My thinking here is about scopes that require MFA for example. If the MFA is verified during SignUp, then if during SignInAfterSignUp we require MFA again for a scope that's a double verification and I thought the user only has to do only once. My point is that the SingInAfterSignUp call will always fail with browser required even after we verified the MFA at a previous step.   
This is not only about MFA but rather any scope that is not automatically granted but user has to grant access via otp. In the case of the SignInWithPassword for example, this would work as once /initiate or /challenge is called it would return that OTP is required. 
For example if the developer asks for the Sharepoint read files but indeed we don't support this as part of Naive Auth but don't want it to be a problem in the future
<img width="1052" alt="Screenshot 2023-12-07 at 15 27 12" src="https://github.com/AzureAD/microsoft-authentication-library-for-objc/assets/111577419/c4a8e5fe-659b-438b-a059-b6915de33ed8">

Reply:
About MFA, I don't know if we can base conditional access on the scopes specified during the signIn, I never heard about it, but it can be. Anyway, MFA will be done sometime next year. About email verification, that is related to the tenant configuration (for signUp). 
Scopes are permissions you want included in the access token received, so they're only related to signIn

Reply:
Ok, then that won't be a problem and most likely we will add a flow from the token endpoint to initiate to challenge back to token at some point which will cater for this scenario.
Will resolve.